### DISCUSSION TITLE
Registry and scheduler-docker interaction

### DISCUSSION THREAD

**Discussion by francipvb** [dokku/dokku#6905] [Q&A]
Hello all,

We use dokku for development deployments. However we have an issue when we do a docker cleanup and all dokku/* images are removed.

I configured dokku to push images to a registry and I would like to tell dokku to pull images from the configured registry when application starts but I didn't found a way to do it, so I have errors when starting up applications after a dokku restart.

I were reading the dokku code for the registry plugin but the code is somewhat complex for my golang knowledge to understand what happens. I was trying to change the docker scheduler to pull the image from the registry but I cann't even start implementing the feature.

Thanks for your help.

**Comment by josegonzalez** [comment]
When you say you have issues after a dokku restart, what do you mean by that? When are you removing images? Dokku shouldn't be doing that for anything in use.

**Reply by francipvb** [reply]
We have dokku deployed in an EC2 instance and we reboot it everiday. The other thing is that we don't have disk space and therefore we do a cleanup of the docker cache and old images, and some dokku images are removed in this process, including application images.

Because of this I ask you about an automatic registry pull of an image if the image is missing. The dokku registry push hook already sets a tag and it is saved to the app configuration.

**Reply by josegonzalez** [reply]
This sounds like a feature request or bug. Mind filing a ticket for this?
